s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/FileEntriesEventEmitter.java

```diff
@@ -103,23 +102,43 @@ private String returnNothingOrThrowExceptionWhenEmissionFailedCompletely(
 
     private void storeErrorReportsInS3(Try<List<PutEventsResult>> failedEntries, ImportRequest input) {
         S3Driver s3Driver = new S3Driver(s3Client, input.extractBucketFromS3Location());
-        UriWrapper reportFilename = generateErrorReportUri(input);
-        List<PutEventsResult> putEventsResults =
-            failedEntries.orElse(this::generateReportIndicatingTotalEmissionFailure);
+        UriWrapper reportFilename = generateErrorReportUri(input, failedEntries);
+        List<PutEventsResult> putEventsResults;
+        putEventsResults =
+            failedEntries.orElse(fails -> generateReportIndicatingTotalEmissionFailure(fails,
+                                                                                       input.getS3Location()));
 
         if (!putEventsResults.isEmpty()) {
             String reportContent = PutEventsResult.toString(putEventsResults);
             s3Driver.insertFile(reportFilename.toS3bucketPath(), reportContent);
         }
     }
 
-    private UriWrapper generateErrorReportUri(ImportRequest input) {
-        UriWrapper inputUri = new UriWrapper(input.getS3Location());
+    private UriWrapper generateErrorReportUri(ImportRequest input, Try<List<PutEventsResult>> failedEntries) {
+        UriWrapper inputUri = new UriWrapper(input.extractPathFromS3Location().toString());
         UriWrapper bucket = inputUri.getHost();
 
         return bucket
                    .addChild(ERRORS_FOLDER)
-                   .addChild(inputUri.getPath());
+                   .addChild(removeLastPartOfPath(inputUri.getPath().toString()))
+                   .addChild(failedEntries.isSuccess() ? "PartialFailure"
+                                 : failedEntries.getException().getClass().getSimpleName())
+                   .addChild(makeFileEndingError(retrieveLastPartOfPath(inputUri.getPath().toString())));
+    }
+
+    private String removeLastPartOfPath(String path) {
+        int lastDeliminatorIndex = path.lastIndexOf('/');
+        return path.substring(0, lastDeliminatorIndex);
+    }
+
+    private String retrieveLastPartOfPath(String path) {
+        int lastDeliminatorIndex = path.lastIndexOf('/');
+        return path.substring(lastDeliminatorIndex);
+    }
+
+    private String makeFileEndingError(String filename) {
+        int lastDot = filename.lastIndexOf('.');
```

Comment:
int extensionSeparatorIndex  = …(EXTENSION_SEPARATOR), although "." is only a separator in this sense in the last position…